utils/sl/comp_build.sh

```diff
@@ -0,0 +1,405 @@
+                             interactive builds.  In this case, the default
+                             build number for each dependence is pulled from
+                             the latest good <component>-update-scratch
+                             build.
+
+       If the default build number for a dependence needs to be updated (due
+       to an API change, for example), then the user can specify <build_number>
+       for that component to a value larger than the default.   The script
+       will always choose the larger of the two values.
+
+CUSTOM
+       If a custom script is specified on the command line, the script will be
+       sourced, if and only if the the build and install steps are successful.
+       Before sourcing the script, COMP_PREFIX is set to the installation
+       directory of the component.
+
+EXAMPLE
+       Building IOF, mcl, DAOS, and CPPR from the job script:
+       This script normalizes the build between various components.  It is no
+       longer necessary to copy boiler plate code into all of the components.
+       This should make things more maintainable.
+
+       The Jenkins job can be updated for each component to something like this
+
+           cd <comp>
+           scons_local/comp_build.sh comp_test.sh
+
+       As an example, let's use CPPR
+       In the case of CPPR, let's say it depends on at least iof build 33, mcl
+       build 27, ompi build 88, and mercury build 123, we would create a config
+       file, call it build.config,  with the following:
+
+           component=cppr
+           depends=iof:33
+           depends=mcl:27:iof
+           depends=ompi:88:iof
+           depends=mercury:123:iof
+
+       we would then create comp_test.sh with the following lines:
+
+           scons utest
+           scons utest --utest-mode=memcheck
+           #Helgrind appears to be unstable
+           scons utest --utest-mode=helgrind
+
+           ${COMP_PREFIX}/TESTING/scripts/run_test.sh -l 4
+           #Run memcheck test
+           ${COMP_PREFIX}/TESTING/scripts/run_test.sh -e m -l 4
+
+       And we would run with the command listed above.
+
+       Notice the <parent_name> for mcl, ompi, and mercury is set to iof.
+       This is for the cppr-update-scratch job to tell it to only use these
+       components that have been blessed by a successful iof-update-scratch
+       job.   For review or interactive building, it will always default to
+       the version blessed by the latest cppr-update-scratch job.  In both
+       cases, the <build_number> can force a newer version of a component.
+EOF
+  exit 1
+}
+
+function print_status()
+{
+    echo "*******************************************************************"
+    echo $*
+    echo "*******************************************************************"
+}
+
+function setup_dep()
+{
+  temp_ifs=$IFS
+  IFS=':'
+  array=($1)
+  IFS=$TEMP_IFS
+
+  len=${#array[@]}
+  if [ $len -lt 1 ] || [ $len -gt 3 ]; then
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]